modelexpress_client/python/modelexpress/heartbeat.py

```
@@ -117,8 +150,41 @@ def _update_status(self, status: int) -> None:
+                    f"Worker will continue without P2P serving."
+                )
+                self._publish_given_up = True
+            return
+
+        try:
+            self._mx_source_id = self._publish_fn()
+            logger.info(
+                f"[Worker {self._worker_rank}] Metadata published successfully "
+                f"(mx_source_id={self._mx_source_id})"
+            )
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)